File: SimulandoHerenciaMultipleRuby/lib/figura_geometrica.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

#juega el papel de clase Abstracta
class FiguraGeometrica

attr_accessor :numLados

  def initialize(lados)
      @numLados=lados
  end  
  
private_class_method :new

  #metodos Abstractos no implementados
  def perimetro                 
    raise NotImplementedError.new
  end
  
  def area
    raise NotImplementedError.new
  end
  
  def to_s
  "\nSoy una figura geometrica de #{@numLados} lados"
end 
  
  
end#class

File: HerenciaMultipleJava/src/herenciamultiplejava/FiguraGeometrica.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package herenciamultiplejava;
/**
 *
 * @author Fco Manuel Guerrero Jimènez
 */
public abstract class FiguraGeometrica {  //Class Abstracta
    
    private int numLados;
    //constructor    
    public FiguraGeometrica(int lados){
        this.setNumLados(lados);
    }
    
    public abstract double perimetro(); //metodo Abstracto
    public abstract double area();      //metodo Abstracto


    public void setNumLados(int numL){
        numLados = numL;
    }
    
    public int getNumLados(){
        return numLados;
    }

public String toString() {
    return "Soy una Figura Geometrica de "+ getNumLados()+ " lados"; //To change body of generated methods, choose Tools | Templates.
    }
}